### PR TITLE
Test glitches

### PR DESCRIPTION
**Pull request by ellert**
# This Pull request:

## Changes or fixes:

Fix some issues with test dound in 6.30.00

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

File: bindings/pyroot/pythonizations/test/rtensor.py [hunk 1]
```diff
@@ -28,8 +29,10 @@ def test_dtypes(self):
         """
         Test adoption of numpy arrays with different data types
         """
-        #using long long on Linux does not work although it is same size as long
-        if (platform.system() == "Windows") :
+        # Using long long on 64 bit Linux does not work although it is
+        # same size as long
+        if ((platform.system() == "Windows") or
+            ((platform.system() == "Linux") and (sys.maxsize < 2**32))) :
```
Comment (guitargeek) [inline bindings/pyroot/pythonizations/test/rtensor.py:35]: What if we scrap this extra check to alter the `cpptypes`, and directly define them to be what they are?

```c++
    cpptypes = ["int32_t", "int64_t", "uint32_t", "uint64_t", "float", "double"]
```
At least this is what I understand from the numpy docs:
https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.int64

File: bindings/pyroot/pythonizations/test/rdataframe_makenumpy.py [hunk 2]
```diff
@@ -147,7 +147,7 @@ def test_sliced_array(self):
         """
         Test correct reading of a sliced numpy array (#13690)
         """
-        table = np.array([[1,2], [3,4]])
+        table = np.array([[1,2], [3,4]], dtype="int64")
```
Comment (guitargeek) [inline bindings/pyroot/pythonizations/test/rdataframe_makenumpy.py:150]: Similar to the other test, should we maybe better use `df.Take['int64_t']` further down to be consistent with NumPy?

**Review by guitargeek — CHANGES_REQUESTED** [review]
Thanks @ellert. I just have some suggestions about the integer types.

File: tutorials/CMakeLists.txt [hunk 3]
```diff
@@ -74,7 +74,8 @@ if(NOT ROOT_minuit2_FOUND)
 endif()
 
 if(NOT clad)
-  set(clad_veto fit/minuit2GausFit.C)
+  set(clad_veto fit/minuit2GausFit.C
+                fit/exampleFit3D.C)
```
[the remark this answers is not in the document]
Comment (ellert) [reply tutorials/CMakeLists.txt:78]: There is no request for changes here, so I will resolve this.

**Comment by guitargeek** [comment]
Hi @ellert, sorry that my suggestions made things even worse and the tests don't pass anymore.

I'll revive this PR myself, updating the commits as necessary before merging and backporting.

File: bindings/pyroot/pythonizations/test/rtensor.py [hunk 4]
```diff
@@ -11,40 +12,54 @@ def check_shape(root_obj, np_obj):
...
     """
 
     # Helpers
-    dtypes = [
-        "int32", "int64", "uint32", "uint64", "float32", "float64"
-    ]
-    cpptypes = ["int", "long", "unsigned int", "unsigned long", "float", "double"]
+    cpptypes = {
+        "int32": "int32_t",
+        "int64": "int64_t" if platform.system() != "Darwin" else "long",
```
Comment (guitargeek) [inline bindings/pyroot/pythonizations/test/rtensor.py:23]: I don't understand why the obvious mapping of `int64` to `int64_t` doesn't work on macOS, but on Windows and Linux it works fine.

Therefore, I'm treating macOS/Darwin as the special case now.

**Review by guitargeek — APPROVED** [review]
Thanks for the initiative! I'll merge this once the CI is green (the backport is also ready).

Please let us know with a GitHub issue or a new PR if there should be still failures on Linux 32 bit in your case.

**Pull request merged**